File: src/tafqeetj/exceptions/NumberOutOfRangeException.java
package tafqeetj.exceptions;

import java.math.BigDecimal;
import java.math.BigInteger;

public class NumberOutOfRangeException extends RuntimeException {
    public NumberOutOfRangeException(int number) {
        super("Out of range value: " + number);
    }

    public NumberOutOfRangeException(byte number) {
        super("Out of range value: " + number);
    }

    public NumberOutOfRangeException(short number) {
        super("Out of range value: " + number);
    }

    public NumberOutOfRangeException(long number) {
        super("Out of range value: " + number);
    }

    public NumberOutOfRangeException(BigInteger number) {
        super("Out of range value: " + number);
    }

    public NumberOutOfRangeException(BigDecimal number) {
        super("Out of range value: " + number.toPlainString());
    }
}

File: src/main/kotlin/tech/moussa/tafqeetk/exceptions/NumberOutOfRangeException.kt
package tech.moussa.tafqeetk.exceptions

import java.math.BigDecimal
import java.math.BigInteger

class NumberOutOfRangeException(
    number: Long
) : RuntimeException("Out of range value: $number") {
    constructor(number: BigInteger) : this(number.toLong())
    constructor(number: BigDecimal) : this(number.toLong())
    constructor(number: Double) : this(number.toLong())
}